crates/dojo/world/src/metadata/ipfs_service.rs

```diff
@@ -0,0 +1,71 @@
+use std::io::Cursor;
+
+use anyhow::Result;
+#[cfg(test)]
+use futures::TryStreamExt;
+use ipfs_api_backend_hyper::{IpfsApi, TryFromUri};
+
+use super::metadata_service::MetadataService;
+
+/// IPFS implementation of MetadataService, allowing to
+/// upload metadata to IPFS.
+pub struct IpfsMetadataService {
+    client: ipfs_api_backend_hyper::IpfsClient,
+}
+
+// impl required by clippy
+impl std::fmt::Debug for IpfsMetadataService {
+    fn fmt(&self, _f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
+        Ok(())
```

Comment:
You could print some accessible data from `IpfsClient`, if any.

Reply:
It looks like nothing is accessible from `ipfs_api_backend_hyper::IpfsClient` 😕